receiver/snowflakereceiver/documentation.md

```diff
@@ -0,0 +1,84 @@
+| snowflake.rows_inserted.total | Number of rows inserted into a table (or tables). | {rows} | Gauge(Int) | <ul> <li>schema_name</li> <li>execution_status</li> <li>error_message</li> <li>query_type</li> <li>warehouse_name</li> <li>database_name</li> <li>warehouse_size</li> </ul> |
+| snowflake.rows_produced.total | Total number of rows produced by statement. | {rows} | Gauge(Int) | <ul> <li>schema_name</li> <li>execution_status</li> <li>error_message</li> <li>query_type</li> <li>warehouse_name</li> <li>database_name</li> <li>warehouse_size</li> </ul> |
+| snowflake.rows_unloaded.total | Total number of rows unloaded during data export. | {rows} | Gauge(Int) | <ul> <li>schema_name</li> <li>execution_status</li> <li>error_message</li> <li>query_type</li> <li>warehouse_name</li> <li>database_name</li> <li>warehouse_size</li> </ul> |
+| snowflake.rows_updated.total | Total number of rows updated in a table. | {rows} | Gauge(Int) | <ul> <li>schema_name</li> <li>execution_status</li> <li>error_message</li> <li>query_type</li> <li>warehouse_name</li> <li>database_name</li> <li>warehouse_size</li> </ul> |
+| snowflake.session_id.count | Distinct session id's associated with snowflake username. | 1 | Gauge(Int) | <ul> <li>user_name</li> </ul> |
+| snowflake.storage.failsafe_bytes.total | Number of bytes of data in Fail-safe. | By | Gauge(Int) | <ul> </ul> |
+| **snowflake.storage.stage_bytes.total** | Number of bytes of stage storage used by files in all internal stages (named, table, user). | By | Gauge(Int) | <ul> </ul> |
+| **snowflake.storage.storage_bytes.total** | Number of bytes of table storage used, including bytes for data currently in Time Travel. | By | Gauge(Int) | <ul> </ul> |
+| **snowflake.total_elapsed_time.avg** | Average elapsed time. | s | Gauge(Double) | <ul> <li>schema_name</li> <li>execution_status</li> <li>error_message</li> <li>query_type</li> <li>warehouse_name</li> <li>database_name</li> <li>warehouse_size</li> </ul> |
+| snowflake.total_elapsed_time.total | Total elapsed time. | s | Gauge(Int) | <ul> <li>schema_name</li> <li>execution_status</li> <li>error_message</li> <li>query_type</li> <li>warehouse_name</li> <li>database_name</li> <li>warehouse_size</li> </ul> |
+
+**Highlighted metrics** are emitted by default. Other metrics are optional and not emitted by default.
+Any metric can be enabled or disabled with the following scraper configuration:
+
+```yaml
+metrics:
+  <metric_name>:
+    enabled: <true|false>
+```
+
+## Resource attributes
+
+| Name | Description | Type |
+| ---- | ----------- | ---- |
+| snowflake.account.name | The name of the snowflake account being monitored. | String |
+| snowflake.username | The name of the snowflake user account being used. | String |
+| snowflake.warehouse.name | The name of the warehouse being used. | String |
```

Comment:
I see these resource attributes are added to some metrics as metrics attributes. We should have either resource attributes or metric attributes, not both.

Resource attributes are supposed to be applied to all of the metrics. If we cannot apply them to all metrics, we should keep metric attributes instead and remove these resource attributes.

Reply:
This requires a little context: The way that snowflake reports these statistics is by running a query. In order to do this via a client call however, an organization is required to define a warehouse (compute resources) a database to search (the default location of the logs is the SNOWFLAKE database) and a user with the appropriate permissions. 

The resource attributes as defined here are meant to reflect those values (as they would be defined in the config) and are not necessarily the same as the metric attributes with similar names. They are also the same for for every metric being generated by any particular collector.

I'm going to update the descriptions to better reflect this sentiment, but if you still don't think that these attributes should be reported let me know!

Reply:
Ok, in order to decide if we need to include these resource attributes we need to answer couple of questions:

- Can the user have another warehouse with different permissions to query the same data? 
- How important that information is to be included in the generated metrics? Does the presence of these resource attributes justify possible confusion with metric attributes? Keep in mind that many exporters don't pass the description field and resource attributes are merged with metrics attributes, e.g. that's the case for signalfx exporter.

Reply:
And resource attributes are supposed to be applied to all the metrics as static attributes, but AFAIU these ones are applied to some "proxy" warehouse that just stores the usage statistics about all other warehouses. Please correct me if my understanding is wrong. If that's the case, I don't think `snowflake.username` and `snowflake.warehouse.name` should be added to resource attributes. But `snowflake.account.name` is a good example of resource attribute because it can be clearly applied to all the metrics.

Reply:
So just to clarify [warehouses](https://docs.snowflake.com/en/user-guide/warehouses-overview.html) are what snowflake calls their compute units. All queries must be associated with a warehouse, and they really don't have much to do with storing data (i know...). So when snowflake.warehouse.name is applied what its really referring to is the compute unit associated with the stats queries we're scraping. 

However! The more I think about it the more I think it might not really matter what the compute resource is named, so I'm all in favor of dropping `snowflake.warehouse.name` altogether. That said, I can imagine scenarios where wanting to know the username and account names associated with these metrics. Although if you're really passionate about removing username I won't argue.

Reply:
> The original justification for these resource attributes was to handle the case where multiple, separate snowflake deployments (say from two different business units within the same company) are being monitored in the same tool by the metric users.

In this case, each different business unit will be uniquely identified by an account name, right?

Reply:
Yea exactly! Having a user to chase down could also be useful but yes ultimately I think account name should be enough.

Reply:
Ok great! Anything else?

Reply:
I see you updated the metadata.yaml, but you need to run `make generate` to update the generated files . Also please remove the unrelated changes as I mentioned in another comment